hack/vendor.sh

```diff
@@ -12,4 +12,15 @@ if ! hash vndr; then
 	exit 1
 fi
 
+if [ $# -eq 0 ]; then
+	# update our vendored copy of archive/tar when doing a full vendor
+	GO_VERSION="${GO_VERSION:=$(grep "ARG GO_VERSION" ./Dockerfile | awk -F'=' '{print $2}')}"
+	rm -rf vendor/archive
+	mkdir -p ./vendor/archive
+	git clone -b go"${GO_VERSION}" --depth=1 git://github.com/golang/go.git ./go
+	git -c user.email="moby@example.com" -c user.name="moby" --git-dir ./go/.git --work-tree ./go am ../patches/0001-archive-tar-do-not-populate-user-group-names.patch
+	cp -a go/src/archive/tar ./vendor/archive/tar
+	rm -rf ./go
```

Comment:
This feels kind of odd to me -- IMO, this would be cleaner using the released source tarballs, since we don't care about history or anything anyhow, something like:

```suggestion
	mkdir -p ./vendor/archive/tar
	curl -fsSL "https://golang.org/dl/go${GO_VERSION}.src.tar.gz" \
		| tar --extract --gzip --directory=vendor/archive/tar --strip-components=4 go/src/archive/tar
	patch --strip=4 --directory=vendor/archive/tar --input="$PWD/patches/0001-archive-tar-do-not-populate-user-group-names.patch"
```